tests/tests_api/test_group.py

```diff
@@ -406,3 +366,48 @@ def test_abort_appends_former_entity_lists(self, api_client, active_group, avail
         assert active_group.students_former.filter(pk=student.pk).exists()
         assert active_group.teachers_former.filter(pk=teacher.pk).exists()
         assert active_group.coordinators_former.filter(pk=coordinator.pk).exists()
+
+
+class TestGroupCreation:
+    def get_data(self, group: Group):
+        return {
+            "lesson_duration_in_minutes": 30,
+            "communication_language_mode": group.communication_language_mode,
+            "availability_slots_for_auto_matching": [
+                a.id for a in group.availability_slots_for_auto_matching.iterator()
+            ],
+            "language_and_level": group.language_and_level.id,
+            "monday": "10:00:00",
+            "coordinators": [c.pk for c in group.coordinators.iterator()],
+            "students": [s.pk for s in group.students.iterator()],
+            "teachers": [t.pk for t in group.teachers.iterator()],
+        }
+
+    def test_group_create_general(self, api_client, group: Group, timestamp):
+        data_to_create = self.get_data(group)
+        response = api_client.post("/api/public/groups/", data_to_create)
+        assert response.status_code == status.HTTP_201_CREATED
+        created_group = Group.objects.get(id=response.data["id"])
+        created_group_serializer = GroupWriteSerializer(created_group)
+        for field, val in data_to_create.items():
+            assert val == response.data[field]
+            assert val == created_group_serializer.data[field]
+
+        assert group.status == GroupStatus.PENDING
+
+        common_status_since = created_group.status_since
+        compare_date_time_with_timestamp(common_status_since, timestamp)
+
+        for student in group.students.iterator():
+            assert student.status == StudentStatus.GROUP_OFFERED
+            assert student.status_since == common_status_since
+            log_event: StudentLogEvent = StudentLogEvent.objects.get(student_id=student.pk)
+            assert log_event.type == StudentLogEventType.GROUP_OFFERED
```

Comment:
@Auch-Auch, after discussion in #191 ([comment](https://github.com/samanthasgroup/django-webapps/pull/191#discussion_r1271420281)) I guess that these lines are where `group_to` should have been checked and the test should have failed because LogEventCreator was creating student and teacher events incorrectly.

Same for the `for teacher ...` loop below.